lua/nvim-lsp-installer/notify.lua

```diff
@@ -8,6 +8,12 @@ return function(msg, level)
             title = TITLE,
         })
     else
-        vim.notify(("[%s] %s"):format(TITLE, msg), level)
+        local vim_log_levels = vim.tbl_keys(vim.log.levels)
+        local log_levels = vim.tbl_map(function(lv)
+            return lv:lower()
+        end, vim_log_levels)
+
+        local Log = require "nvim-lsp-installer.log"
+        Log[log_levels[level]](msg)
```

Comment:
I'm not sure this needs changing. In my mind, notifications to the user (this module) are very different than system logging (my idea of `log.lua`).

[`vim.notify()` already takes care of setting highlight group based on log level](https://github.com/neovim/neovim/blob/master/src/nvim/lua/vim.lua#L421-L430), so I think going through the log is redundant here, no?

Reply:
Keep in mind, you can create another instance of the logger that doesn't write to a file. 

This API is a little bit more flexible, unless you're very set on the use-case of `notify`, with the main benefit being that it allows three different modes: { 'sync', 'async', false }.

Reply:
I don't have any strong opinions (or well, I might but they're loosely held) on this. I fail to see the reason to substitute usage of the native `vim.notify` for this logger, for messages intended to be displayed to the user - it just adds overhead with no benefits unless I am missing something. As for "system" logs stored to file, I definitely see a need for a logger implementation like this, but again I'd like to keep these separate from one another.

What I'm picturing is 
1. keeping the `notify` module as-is (i.e., keep going through `vim.notify` directly for message intended for users)
2. change the `log` module to only act as a system logger by default (no `echo`, only `io.write`)

Am I missing something here or does that make sense? I'd be happy to help contribute with above changes and at the meantime add some more logging + a new `vim.g` option to control the system log level + docs.

Reply:
Sounds good, it's an easy swap like I said. 

The only thing really that's left in this case is deciding on some kind of method to set the log-level. You can always use a global variable with `vim.g`, but I also think it might be worth it to consider using an `init()` function soon.

Reply:
I think a global variable, that can be changed and have direct impact throughout a neovim session, is pretty nice for this particular setting, no? To temporarily increase logging in order to debug something for example. No need to change your own Lua code. I was never a fan of plugins having `setup()` or `init()` functions tbh, but is that where the direction is generally going?

Reply:
> edit: Ah the logger impl doesn't allow changing log level after instantiation

Yeah that's the problem, so it's not a toggle. That's why I figured it would make sense to have in a setup function. I'll try something else and see if it works.

Reply:
I noticed earlier that it seems like the lua code executes before `plugin/nvim-lsp-installer.vim` (where defaults are defined). That caught me a bit off guard tbh.

Reply:
I think it should be fine now. I tested with with changing the log-level at runtime and it worked correctly for me 😃 

This is now a hybrid between the plenary logger and the one used by the LSP module in core.